crd-generator/api/src/test/java/io/fabric8/crd/generator/CRDGeneratorTest.java

```diff
@@ -578,32 +575,6 @@ void checkK8sValidationRules() throws Exception {
     assertTrue(outputDir.delete());
   }
 
-  private void assertFileEquals(final File expectedFile, final File actualFile) {
```

Comment:
Since we are at it, would it make sense use approval tests?
We are using them in the `java-generator`.

Reply:
I'm not sure if I understand you correct. What do you mean with "approval tests"?

My general intention is to reduce the size of the CRDGeneratorTest to make it easier to write more tests.

To make this happen, I started with extracting some assertion methods to the utility class `CRDGeneratorAssertions`.

Then I added `CRDGeneratorExamplesTest` to group all "end to end" tests.
We can add here a test for each of our examples in the future.

This `CRDGeneratorExamplesTest` class is also extended by `ParallelCRDGeneratorExamplesTest` , so that every test written in `CRDGeneratorExamplesTest` is also tested with the "parallel generation" feature.


I have looked into the java-generator tests and found `io.fabric8.java.generator.testing.KubernetesResourceDiff`. I was not aware of it but if I look a little bit closer at it now, it reads YAML files to an internal representation and compares this. In conclusion `KubernetesResourceDiff` can compare the syntax of a two yaml files but it cannot ensure that both YAML files are exactly the same (which is important for the deterministic output feature).

Reply:
Approval tests are "Golden Master", if I understand it correctly you are comparing plain `yaml` to files stored in git, and that's the use-case for them.

Dependency:
https://github.com/fabric8io/kubernetes-client/blob/0c80a66d10bf7096864246ce453413e0da4edbd2/java-generator/core/pom.xml#L95

Approval tests:
https://github.com/fabric8io/kubernetes-client/blob/main/java-generator/core/src/test/java/io/fabric8/java/generator/ApprovalTest.java

and here you find the "golden masters":
https://github.com/fabric8io/kubernetes-client/tree/main/java-generator/core/src/test/resources/io/fabric8/java/generator/approvals

I think that using a similar infrastructure in the `crd-generator` might help in your noble goal of:

> reduce the size of the CRDGeneratorTest to make it easier to write more tests

Reply:
Thanks for the hints. I didn't know the project [ApprovalTests](https://github.com/approvals/ApprovalTests.Java) and yes it looks like it can replace the current self-developed assertions.

https://github.com/fabric8io/kubernetes-client/issues/5850

Reply:
Nice!
Thanks a lot for taking the time to make the improvement 👍 let's discuss approvals in the other PR.